sentry_sdk/integrations/django/__init__.py

```diff
@@ -665,12 +666,21 @@ def _set_db_data(span, cursor_or_db):
     vendor = db.vendor
     span.set_data(SPANDATA.DB_SYSTEM, vendor)
 
-    connection_params = (
-        cursor_or_db.connection.get_dsn_parameters()
-        if hasattr(cursor_or_db, "connection")
+    if (
+        hasattr(cursor_or_db, "connection")
         and hasattr(cursor_or_db.connection, "get_dsn_parameters")
-        else db.get_connection_params()
-    )
+        and inspect.isfunction(cursor_or_db.connection.get_dsn_parameters)
```

Comment:
Shouldn't be, that's a [psycopg2 sync function](https://github.com/search?q=repo%3Apsycopg%2Fpsycopg2%20get_dsn_parameters&type=code).

Btw, sidenote to this. [Newer Django](https://docs.djangoproject.com/en/4.2/ref/databases/#postgresql-notes) uses psycopg (i.e., psycopg 3) instead of psycopg2. psycopg doesn't have this function. So this condition will always fall back to using `get_connection_params` instead. This is relevant because we [added](https://github.com/getsentry/sentry-python/pull/2308) the `get_dsn_parameters` part to bypass calling `get_connection_params` for postgres.

Reply:
My hunch was correct 😞 Reopened the original issue: https://github.com/getsentry/sentry-python/issues/2303